application/server/server.go

```diff
@@ -111,6 +111,16 @@ func textDocumentDidChangeHandler() jrpc2.Handler {
 
 		di.FileWatcher().SetFileAsChanged(params.TextDocument.URI)
 
+		filePath := uri.PathFromUri(params.TextDocument.URI)
+		autoScanEnabled := c.IsAutoScanEnabled()
+
+		f := workspace.Get().GetFolderContaining(filePath)
+
+		if f != nil && autoScanEnabled && uri.IsDotSnykFile(params.TextDocument.URI) {
+			go f.ScanFolder(ctx)
+			return nil, nil
+		}
+
 		for _, change := range params.ContentChanges {
 			if packageScanner, ok := di.Scanner().(snyk.PackageScanner); ok {
 				packageScanner.ScanPackages(ctx, c, uri.PathFromUri(params.TextDocument.URI), change.Text)
```

Comment:
@bastiandoetsch I also added the trigger when a user runs the command `snyk ignore --id='ISSUE_ID' --reason='REASON'` in the terminal.

I need to commit the unit test for this branch and move this logic to a function so both handlers can use it

Reply:
Oh yeah, probably when VSCode refreshes and the content changes, it is sending this.

Reply:
Let me know, once you've pushed it, I'll re-review and merge.

Reply:
For this case, I'm trying to make this test pass:


```go
func Test_textDocumentDidChangeHandler_shouldTriggerScanForDotSnykFile(t *testing.T) {
	loc, jsonRPCRecorder := setupServer(t)
	config.CurrentConfig().SetSnykCodeEnabled(false)

	fakeAuthenticationProvider := di.AuthenticationService().Provider().(*snyk.FakeAuthenticationProvider)
	fakeAuthenticationProvider.IsAuthenticated = true

	_, err := loc.Client.Call(ctx, "initialize", nil)
	if err != nil {
		t.Fatalf("initialization failed: %v", err)
	}

	snykFilePath, folderPath := createTemporaryDirectoryWithSnykFile(t)

	// Simulate a "did change" event for the .snyk file
	sendFileChangedMessage(t, snykFilePath, folderPath, loc)

	// Wait for $/snyk.scan notification
	assert.Eventually(
		t,
		checkForSnykScan(t, jsonRPCRecorder),
		15*time.Second,
		50*time.Millisecond,
	)
}

func sendFileChangedMessage(t *testing.T, filePath, fileDir string, loc server.Local) sglsp.DocumentURI {
	t.Helper()
	c := config.CurrentConfig()
	didChangeParams := sglsp.DidChangeTextDocumentParams{
		TextDocument: sglsp.VersionedTextDocumentIdentifier{
			TextDocumentIdentifier: sglsp.TextDocumentIdentifier{
				URI: uri.PathToUri(filePath),
			},
			Version: 1,
		},
		ContentChanges: []sglsp.TextDocumentContentChangeEvent{
			{
				Text: "SNYK-JS-QS-3153490", // simulates: `snyk ignore --id='SNYK-JS-QS-3153490'`
			},
		},
	}
	workspace.Get().AddFolder(workspace.NewFolder(c, fileDir,
		"Test",
		di.Scanner(),
		di.HoverService(),
		di.ScanNotifier(),
		di.Notifier()))

	_, err := loc.Client.Call(ctx, "textDocument/didChange", didChangeParams)
	if err != nil {
		t.Fatal(err)
	}

	return didChangeParams.TextDocument.URI
}
```

I'm following a similar approach written for other cases but I wasn't able to solve this error yet:

```
024-06-28T16:07:55+01:00 INF - Fake Authentication - successful.
2024-06-28T16:07:55+01:00 DBG - IsAuthenticated: fake auth successful
2024-06-28T16:07:55+01:00 INF ide.workspace.folder.DelegatingConcurrentScanner.ScanFile  - Scan was canceled err={}
--- FAIL: Test_textDocumentDidChangeHandler_shouldTriggerScanForDotSnykFile (15.21s)
    server_test.go:1033: 
        	Error Trace:	/Users/cata/git/snyk/hammerhead/snyk-ls/application/server/server_test.go:1033
        	Error:      	Condition never satisfied
        	Test:       	Test_textDocumentDidChangeHandler_shouldTriggerScanForDotSnykFile
FAIL
```

which is happening here: `domain/snyk/scanner.go`.

We could go for the `textDocumentDidSaveHandler` only and work on the `textDocumentDidChangeHandler` in another PR. WDYT?

Reply:
@bastiandoetsch here is the initial change needed in IntelliJ https://github.com/snyk/snyk-intellij-plugin/pull/562